Qt debugger hotkey fixes

aldelaro5 opened the pull request:
This is a set of 2 bugfixes for the debugger hotkeys:

1. Do not have the hotkeys activate when the debugger is disabled.
2. Do not show the debugging hotkey tab if the debugger is disabled.

leoetlino commented on Source/Core/DolphinQt2/Config/Mapping/MappingWindow.cpp:
```diff
@@ -289,7 +289,22 @@ void MappingWindow::SetMappingType(MappingWindow::Type type)
-    AddWidget(tr("Debugging"), new HotkeyDebugging(this));
+
+    HotkeyDebugging* debugging_widget = new HotkeyDebugging(this);
+    connect(&Settings::Instance(), &Settings::DebugModeToggled, [=](bool enabled) {
```
`connect(&Settings::Instance(), &Settings::DebugModeToggled, this, ...)`

leoetlino commented on Source/Core/DolphinQt2/Config/Mapping/MappingWindow.cpp:
```diff
@@ -289,7 +289,22 @@ void MappingWindow::SetMappingType(MappingWindow::Type type)
+    connect(&Settings::Instance(), &Settings::DebugModeToggled, [=](bool enabled) {
+      if (enabled)
+      {
+        m_tab_widget->insertTab(2, GetWrappedWidget(debugging_widget, this, 150, 150),
```
You should store the previous widget instead of creating a new one every time, because removeTab does not delete the object -- meaning you'll leak a debugging widget + the wrapper every time.

leoetlino commented on Source/Core/DolphinQt2/Debugger/CodeWidget.cpp:
```diff
@@ -247,6 +247,11 @@ void CodeWidget::SetAddress(u32 address, CodeViewWidget::SetAddressUpdate update
   m_code_view->SetAddress(address, update);
 }
 
+void CodeWidget::SetBlockSteps(bool blockSteps)
```
`block_steps`

aldelaro5 commented:
@leoetlino ok so the deal is I removed the third commit because #6879 fixes it more properly so the third comment doesn't apply, but I addressed the other 2 comments.

aldelaro5 commented:
@dolphin-emu-bot rebuild

spycrab commented on Source/Core/DolphinQt2/HotkeyScheduler.cpp:
```diff
@@ -213,29 +213,32 @@ void HotkeyScheduler::Run()
 
-      if (IsHotkey(HK_STEP))
-        emit Step();
+      if (SConfig::GetInstance().bEnableDebugging)
```
Maybe move this out into another function so you can just replace this long indented region with a call to ``CheckNextHotkey()``.
